Dockerfile

```diff
@@ -1,15 +1,20 @@
-FROM node:20.12.2
+FROM node:20.12.2-slim
+
+RUN apt-get update || : && apt-get install -y \
+    python3 \
+    build-essential \
+    jq
 
 ENV ENKETO_SRC_DIR=/srv/src/enketo
 WORKDIR ${ENKETO_SRC_DIR}
 
 COPY . ${ENKETO_SRC_DIR}
 
-# Install (including dev dependencies) and build
-RUN yarn install --frozen-lockfile
-
-# Remove dev dependencies after build
-RUN yarn install --production --frozen-lockfile --ignore-scripts
+# Install and build, first with dev dependencies, then with production only (yarn 1 has no prune)
+RUN yarn install --frozen-lockfile \
+    && yarn remove $(cat package.json | jq -r '.devDependencies | keys | join(" ")') -W \
```

Comment:
This step wasn't present previously; has something changed?  It also looks like it might modify `yarn.lock`, and so potentially cause the `yarn install --production` to resolve dependencies differently to those expected...

Reply:
It doesn’t work for reasons I don’t entirely understand. It looks like the native dependencies don’t get rebuilt for some reason.

Reply:
because `--ignore-scripts`?

Reply:
Maybe so. But the other thing is that I’d rather we not have to rebuild them. It’s time consuming and feels generally silly to do that work multiple times.

Reply:
@alxndrsn directly shared with me a concern that `yarn remove` might rebuild the whole lock file. I've confirmed that `yarn remove` surgically removes the dependency(ies) specified.

Reply:
I think the `yarn remove` step is safe but it certainly is not the most elegant thing. Let's remove for now and keep dev dependencies. It still halves the image size and the dev dependencies were included before anyway.